autobahn/twisted/wamp.py

```diff
@@ -84,7 +85,29 @@ def _reject_future(future, error):
 
     @staticmethod
     def _add_future_callbacks(future, callback, errback):
-        return future.addCallbacks(callback, errback)
+        # callback and/or errback may be None
+
+        # Note that errback takes 3 args: type, exception, tb our
+        # errbacks can be consistent between Twisted and asyncio.
```

Comment:
Ah, I see. So that's the issue with the current code even: with Twisted, app code gets `Failure` instead of plain exception. It should be the latter for both. And yes, that requires a Twisted wrapper to unfold the plain exception value. Regardng naming: sorry, yes, the naming suggest private, but it's actually intended as incubation to be made public. Because if apps want to write source compatible code (between Twisted and asyncio), they would be pointed to that API - if we release it.